ast-checks/src/bin/no-uri-literals.rs

```diff
@@ -0,0 +1,43 @@
+use codespan::{FileId, Files};
+use nixpkgs_hammering_ast_checks::analysis::*;
+use nixpkgs_hammering_ast_checks::common_structs::{NixpkgsHammerMessage, SourceLocation};
+use rnix::SyntaxKind::*;
+use std::{env, error::Error};
+use nixpkgs_hammering_ast_checks::tree_utils::walk_kind;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args: Vec<String> = env::args().skip(1).collect();
+    analyze_files(args, analyze_single_file)
+}
+
+fn analyze_single_file(
+    files: &Files<String>,
+    file_id: FileId,
+) -> Result<Report, Box<dyn Error>> {
+    let root = find_root(files, file_id)?;
+    let mut report: Report = vec![];
+
+    // Find all URI literals and report them.
+    for uri in walk_kind(&root, TOKEN_URI) {
+        let start = uri.text_range().start().to_usize() as u32;
+        let loc = files.location(file_id, start)?;
+
+        report.push(NixpkgsHammerMessage {
+            cond: true,
```

Comment:
I put `cond` in here to have consistency with the JSON that the overlays-based checks produced, but perhaps we should remove it from the json for both. It's not useful to the end-user since it's always `true`.

Reply:
The `cond` is used in the Nix checks since otherwise we would have to use `lib.optional` and then `concat` them. Probably not a big issue but it felt nicer at the time.

In Rust we should probably use `Option` and then filter `None`s out in that use case so dropping `cond` here makes sense.